src/error.rs

```diff
@@ -27,9 +29,11 @@ pub enum ErrorKind {
     /// Protocol versions did not match
     ProtocolVersionMismatch,
     /// Could not send on `SendChannel`.
-    SendError(SendError<SocketEvent>),
+    SendError(SendError<Either<ConnectionEvent<SendEvent>, ConnectionEvent<ReceiveEvent>>>),
```

Comment:
It is a send error from the point of view of crossbeam crate, and it is used for sending events and receiving events. `SocketEvent` was actually renamed to `ReceiveEvent` because I introduced new type, `SendEvent`. So the way it works, hasn't changed.
I do agree that it sounds confusing, but what should I rename to
`ChannelSenderError(SendError<Either<ConnectionEvent<SendEvent>, ConnectionEvent<ReceiveEvent>>>` ?

Reply:
changed to `ChannelSenderError(SendError<Either<ConnectionEvent<SendEvent>, ConnectionEvent<ReceiveEvent>>>)`

Reply:
Changed to 
```rust
ChannelSendingError(SendError<ConnectionEvent<SendEvent>>),
ChannelReceivingError(SendError<ConnectionEvent<ReceiveEvent>>),
```